services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java

```diff
@@ -177,6 +179,18 @@ else if (portId == null && subnetId != null) {
         }
         subnet.setAttachedRouterId(routerId);
 
+        List<String> gwPorts = router.getGatewayPorts();
+        if (gwPorts == null || gwPorts.size() == 0) {
+            router.setOwner(subnet.getVpcId());
+        } else {
```

Comment:
@cj-chung If this is a special handling, let us put some comments here so that it reminds us in the future.